metrics/processors/pod_based_enricher.go

```diff
@@ -46,26 +46,22 @@ func (this *PodBasedEnricher) Process(batch *core.DataBatch) (*core.DataBatch, e
 		return nil, err
 	}
 	for _, pod := range pods {
-		addPodInfo(pod, batch)
+		// Only keep stats for pending & running pods. We keep pending pods as well
+		// as running because pending pods might be partially started (e.g. one container
+		// in a multi-container pod).
+		if pod.Status.Phase == kube_api.PodPending || pod.Status.Phase == kube_api.PodRunning {
+			addPodInfo(pod, batch)
+		}
 	}
 	return batch, nil
 }
 
 func addPodInfo(pod *kube_api.Pod, batch *core.DataBatch) {
-	podMs, found := batch.MetricSets[core.PodKey(pod.Namespace, pod.Name)]
+	podKey := core.PodKey(pod.Namespace, pod.Name)
+	podMs, found := batch.MetricSets[podKey]
 	if !found {
-		// create pod stub.
```

Comment:
Is it possible for cAdvisor that a PodStatsContainer is not present but regular pod containers are? If yes then we should clean up the mess somewhere - either create the missing pod or drop the "orphaned" containers.